### PR TITLE
Feature/project clusters

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Show project clusters

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/ExploreDeep/MapView/index.tsx [hunk 1]
```diff
@@ -259,6 +228,40 @@ function ExploreDeepMapView(props: Props) {
         setClusterClicked(false);
     }, []);
 
+    const createClusterMarker = useCallback((markerProps: object) => {
+        const {
+            project_ids: projectIds,
+        } = markerProps as { project_ids: string | undefined };
```
Comment (tnagorra) [inline app/views/ExploreDeep/MapView/index.tsx]:
```suggestion
        } = markerProps as ClusterProperties;
```

Shouldn't this just be ClusterProperties?

File: app/views/ExploreDeep/MapView/index.tsx [hunk 2]
```diff
@@ -259,6 +228,40 @@ function ExploreDeepMapView(props: Props) {
...
+
+        const uniqueProjects = unique(
+            projectIds?.split(',')?.filter((id) => id.length > 0) ?? [],
```
Comment (tnagorra) [inline app/views/ExploreDeep/MapView/index.tsx]:
```suggestion
            projectIds?.split(',').filter((id) => id.length > 0) ?? [],
```

File: app/views/ExploreDeep/MapView/index.tsx [hunk 3]
```diff
@@ -259,6 +228,40 @@ function ExploreDeepMapView(props: Props) {
...
+            (id) => id,
+        );
+        const uniqueProjectsCount = uniqueProjects.length;
+
+        const width = Math.min(8 * Math.log10(uniqueProjectsCount) + 20, 100);
```
Comment (tnagorra) [inline app/views/ExploreDeep/MapView/index.tsx]: Let's check if uniqueProjectsCount is zero by any chance

File: app/views/ExploreDeep/PublicMapView/index.tsx [hunk 4]
```diff
@@ -249,6 +254,40 @@ function ExploreDeepMapView(props: Props) {
         setClusterClicked(false);
     }, []);
 
+    const createClusterMarker = useCallback((markerProps: object) => {
```
Comment (tnagorra) [inline app/views/ExploreDeep/PublicMapView/index.tsx]: Why are we repeating this map view?

Comment (tnagorra) [reply app/views/ExploreDeep/PublicMapView/index.tsx]: The map should be re-used with the fetching logic separated out.

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by subinasr — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

File: app/views/ExploreDeep/ProjectCountMap/index.tsx [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+                        }
+                    },
+                );
+            }
+
+            if (feature.properties.projectId) {
+                onClickedFeaturePropertiesChange([feature.properties.projectId]);
+            } else {
+                onClickedFeaturePropertiesChange([]);
+            }
```
Comment (tnagorra) [inline app/views/ExploreDeep/ProjectCountMap/index.tsx]: Shouldn't the second if also be on if-else chain?

File: app/views/ExploreDeep/ProjectCountMap/index.tsx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+            if (clusterSource) {
+                clusterSource.getClusterLeaves(
+                    cluster_id,
+                    point_count,
+                    0,
+                    (___, aFeatures) => {
+                        if (aFeatures) {
+                            const projectIds = aFeatures
+                                .map((f) => f?.properties?.projectId)
+                                .filter(isDefined);
```
Comment (tnagorra) [inline app/views/ExploreDeep/ProjectCountMap/index.tsx]: Shouldn't we also take a unique count here?

File: app/views/ExploreDeep/ProjectCountMap/index.tsx [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+            return undefined;
+        }
+        const projectFeatures = projects.map((projectByRegion) => (
+            projectByRegion.projectsId?.map((project) => ({
+                id: project,
+                type: 'Feature' as const,
+                geometry: projectByRegion.centroid as GeoJSON.Point,
+                properties: {
+                    id: project,
+                    projectId: project,
```
Comment (tnagorra) [inline app/views/ExploreDeep/ProjectCountMap/index.tsx]: Why set id and projectId as the same value?

**Review by tnagorra — DISMISSED** [review]

File: app/views/ExploreDeep/ProjectCountMap/index.tsx (hunk 7 above)
Comment (AdityaKhatri) [reply app/views/ExploreDeep/ProjectCountMap/index.tsx]: Sir already there

File: app/views/ExploreDeep/ProjectCountMap/index.tsx (hunk 6 above)
Comment (AdityaKhatri) [reply app/views/ExploreDeep/ProjectCountMap/index.tsx]: Sir already there

File: app/views/ExploreDeep/ProjectCountMap/index.tsx (hunk 5 above)
Comment (AdityaKhatri) [reply app/views/ExploreDeep/ProjectCountMap/index.tsx]: Sir already there

**Pull request merged**